jena-arq/src/main/java/org/apache/jena/sparql/expr/aggregate/AggMode.java

```diff
@@ -0,0 +1,150 @@
+    { 
+        return new AccMode(getExpr()) ;
+    }
+
+    @Override
+    public Node getValueEmpty()     { return NodeValue.toNode(noValuesToMode) ; } 
+    
+    @Override
+    public int hashCode()   { return HC_AggMode ^ getExprList().hashCode() ; }
+
+    @Override
+    public boolean equals(Aggregator other, boolean bySyntax) {
+        if ( other == null ) return false ;
+        if ( this == other ) return true ;
+        if ( ! ( other instanceof AggMode ) ) return false ;
+        AggMode a = (AggMode)other ;
+        return exprList.equals(a.exprList, bySyntax) ;
+    }
+    
+    // ---- Accumulator
+    private static class AccMode extends AccumulatorExpr
+    {
+        // Non-empty case but still can be nothing because the expression may be undefined.
+        private NodeValue total = noValuesToMode ;
+        private int count = 0 ;
+        ArrayList<NodeValue> collection=new ArrayList<NodeValue>(); 
+                
+        public AccMode(Expr expr) { super(expr, false) ; }
+
+        @Override
+        protected void accumulate(NodeValue nv, Binding binding, FunctionEnv functionEnv)
+        { 
+			log.debug("mode {}", nv);
+
+            if ( nv.isNumber() )
+            {
+              count++ ;
+              collection.add(nv);
+            }
+            else
+            {
+                //ARQ.getExecLogger().warn("Evaluation error: mode() on "+nv) ;
+                throw new ExprEvalException("mode: not a number: "+nv) ;
+            }
+
+            log.debug("mode count {}", count);
+        }
+
+        @Override
+        public NodeValue getAccValue()
+        {
+            double mode= Double.NaN;
+            if ( count == 0 ) return noValuesToMode ;
+            if ( super.errorCount != 0 )
+                return null ;
+            
+            int indexsize = collection.size();
+            double[] arrDouble = new double[indexsize];
+            for(int i=0; i<indexsize; i++){
+            	arrDouble[i] = collection.get(i).getDouble();	
+            }
+
+    	    HashMap<Double,Integer> amode = new HashMap<Double,Integer>();
+    	    int max  = 0;
+
+    	    for(int i = 0; i < arrDouble.length; i++) {
+    	        if (amode.get(arrDouble[i]) != null) {
+    	            int count = amode.get(arrDouble[i]);
+    	            count++;
+    	            amode.put(arrDouble[i], count);
+
+    	            if(count > max) {
+    	                max  = count;
+    	                mode = arrDouble[i];
+    	            }
+    	        }
+    	        else {
+    	            amode.put(arrDouble[i],1);
+    	            if(mode==0) {
+    	            	mode = arrDouble[i];
+    	            }
+    	        }
+    	    }
+    	    
+    	    for (Map.Entry<Double, Integer> entry: amode.entrySet())
+    	    {
+    	        if (entry.getValue() == max)
+    	        {
+    	    	    System.out.println("Print all max: "+entry.getKey()+" Count: "+entry.getValue());

```

Comment:
I have removed this part. It belongs to a multimodal mode implementation that I am not going to add as a basic function to ARQ. It might be a good function for a custom stats extension though. good catch.